pkg/database/user_report.go

```diff
@@ -53,6 +55,14 @@ type UserReport struct {
 	UpdatedAt time.Time
 }
 
+func (ur *UserReport) AuditID() string {
+	return fmt.Sprintf("user_report:%d", ur.ID)
+}
+
+func (ur *UserReport) AuditDisplay() string {
+	return fmt.Sprintf("hash: %q claimed: %v", ur.PhoneHash, ur.CodeClaimed)
```

Comment:
This was intentional - if a system admin purged a bunch of phone numbers, this is the only way to recover what they did. If we log the row ID, that requires going to backup to reconstruct, and the reconstruct the HMAC.

We get alerted on someone accessing the HMAC secret as well.

I guess we need to prioritize which thing we want to protect against the most.

Reply:
If an admin did that, I think we would probably care more about the timestamp of the audit and just restore all phone numbers from a backup (ignoring conflicts that already exist). It wouldn't be necessary to access the HMAC key or insert specific phone numbers. I think we're be accessing the backup either way, but logging the row ID makes it less likely to leak the HMAC secret.

Also, keep in mind that the HMAC secret rotates over time, so there's no guarantee that, if we discovered this after a rotation, that the _current_ HMAC secret is the value that the phone number was written as (you'd have to check all HMAC keys), which would be rather toilsome.

Reply:
right, only good for a certain period of time.

before I change to row ID.

This would prevent recovering the phone numbers O(time between backups)

Reply:
Talked live - gonna take the first 6-8 characters for display.

```
abcd1234 (claimed: false)
```